### ISSUE TITLE
[Test Maps] `vindictus`

### ISSUE THREAD

**Issue by snake-biscuits** [snake-biscuits/bsp_tool#40]

**Comment by snake-biscuits** [comment]
As of October 2022 Vindictus is a Live MMO with no known modding community, custom servers or way to even run old maps via console.
I haven't researched much about the modding scene here, but it doesn't seem likely one exists outside cheat development

**Comment by xtruong712** [comment]
hope tools can use for Vindictus, because new Bsp cant run in old client (1.69), it auto crash and dont show log so hard for fix

**Comment by snake-biscuits** [comment]
I didn’t know there was an old client out there, might explain why some vindictive maps aren’t passing tests rn
IIRC I had to update my vindictus script a bit since the format isn’t the same as what’s on VDC

CSO2 also changed their map format (without updating the bsp version) around 2013 so it’s possible Nexon did the same thing with Vindictus

Will have to look into the 1.69 client, that’d definitely help with figuring out if the format changed.
Could even potential lead to backporting maps, or even a compiler.

**Comment by snake-biscuits** [comment]
Found a .vmf in maps/ earlier today
It’d make sense to use Hammer(++)
Asset compatibility simplifies things too

Still need a compiler though

**Comment by xtruong712** [comment]
> Found a .vmf in maps/ earlier today It’d make sense to use Hammer(++) Asset compatibility simplifies things too
> 
> Still need a compiler though

now client 1.69  can use bsp of new Vindictus?

**Comment by xtruong712** [comment]
> Found a .vmf in maps/ earlier today It’d make sense to use Hammer(++) Asset compatibility simplifies things too
> 
> Still need a compiler though

are you know what different bsp of new vin and vin 1.69? i want use BSP of new vin for vin 1.69 i need remove what key for it work?

**Comment by snake-biscuits** [comment]
> are you know what different bsp of new vin and vin 1.69?
not yet
I have separate branch scripts for v1.69 (`nexon.vindictus69`) & Steam (`nexon.vindictus)
The only clear difference is the StaticProp formats

Porting `nexon.vindctus.StaticPropv6` back to `nexon.vindictus.StaticPropv5` is doable
However I don't know which maps those are
The Steam release of Vindictus has a bunch of old v1.69 maps mixed in w/ current maps
There's also a `nexon.vindictus.StaticPropv7` format, which I haven't worked out yet

tl;dr: I could maybe port some maps, but it's hard to tell which

**Comment by snake-biscuits** [comment]
Looks like there's 335 v1.69 maps, 17 SPRPv6 & 122 SPRPv7 maps in Vindictus `maps/` folder
> Updates since ~Nov 2022 might've added more SPRPv7 maps, didn't check

The 17 SPRPv6 maps should be easy to port:
```
17e.bsp
17e_ending.bsp
3_24_ending.bsp
arisha_teaser.bsp
arisha_teaser_b.bsp
h03_cut_b.bsp
lobby_beautyshop_body.bsp
lobby_beautyshop_cichol.bsp
lobby_showcase_ari_runway_01.bsp
lobby_showcase_ari_runway_02.bsp
lobby_showcase_ari_runway_03.bsp
lobby_showcase_ari_runway_04.bsp
lobby_showcase_ari_runway_05.bsp
lobby_showcase_runway_chinatube.bsp
lobby_showcase_silhouette.bsp
lobby_showcase_silhouette_ari.bsp
s3_game_create_character.bsp
```

I haven't looked into launching maps in v1.69 yet though, so I can't test any conversions myself

**Comment by xtruong712** [comment]
@snake-biscuits maybe SPRPv7 crash in v1.69, 17 SPRPv6 can run i think so, i have client v1.69 now, if you want test can send me a file

**Comment by xtruong712** [comment]
> Looks like there's 335 v1.69 maps, 17 SPRPv6 & 122 SPRPv7 maps in Vindictus `maps/` folder
> 
> > Updates since ~Nov 2022 might've added more SPRPv7 maps, didn't check
> 
> The 17 SPRPv6 maps should be easy to port:
> 
> ```
> 17e.bsp
> 17e_ending.bsp
> 3_24_ending.bsp
> arisha_teaser.bsp
> arisha_teaser_b.bsp
> h03_cut_b.bsp
> lobby_beautyshop_body.bsp
> lobby_beautyshop_cichol.bsp
> lobby_showcase_ari_runway_01.bsp
> lobby_showcase_ari_runway_02.bsp
> lobby_showcase_ari_runway_03.bsp
> lobby_showcase_ari_runway_04.bsp
> lobby_showcase_ari_runway_05.bsp
> lobby_showcase_runway_chinatube.bsp
> lobby_showcase_silhouette.bsp
> lobby_showcase_silhouette_ari.bsp
> s3_game_create_character.bsp
> ```
> 
> I haven't looked into launching maps in v1.69 yet though, so I can't test any conversions myself

maps/4_00.bsp
maps/4_07.bsp
maps/4_07.txt
maps/4_01.bsp
maps/4_01.txt
maps/4_00_cut_jh.bsp
maps/4_04.bsp
maps/4_06.bsp
maps/4_05.bsp
maps/4_05.txt
maps/4_03.bsp
maps/4_02.bsp
maps/4_02.entity
maps/w_boss04.bsp
maps/w_boss03.bsp



i think this is maps i want change to SPRPv5 (1.69)

**Comment by snake-biscuits** [comment]
These maps are SPRPv7:
> maps/4_00.bsp
> maps/4_07.bsp
> maps/4_01.bsp
> maps/4_04.bsp
> maps/4_06.bsp
> maps/4_05.bsp
> maps/4_03.bsp
> maps/4_02.bsp

I don't have `w_boss04` or `w_boss03` extracted yet, are they new?
Should have enough hard drive space to grab those, but they're probably also SPRPv7

**Comment by xtruong712** [comment]
[w_boss04.zip](https://github.com/snake-biscuits/bsp_tool/files/12533927/w_boss04.zip)


this is w_boss4

**Comment by snake-biscuits** [comment]
@xtruong712 try this:
[w_boss04.backport.zip](https://github.com/snake-biscuits/bsp_tool/files/12534754/w_boss04.backport.zip)

**Comment by xtruong712** [comment]
![image](https://github.com/snake-biscuits/bsp_tool/assets/72967317/fe324987-ce17-4b56-95d1-ce8033442002)

it good 
can run now, but some error with materials

**Comment by snake-biscuits** [comment]
Good to know that it works
The screenshot looks like an issue with Direct X levels, or maybe missing assets

**Comment by xtruong712** [comment]
> Good to know that it works The screenshot looks like an issue with Direct X levels, or maybe missing assets

i check and my client missing some MDL, i will try add it. can you make a guide for do this? because have more bsp SPRPv7 i want change to old SPRP

**Comment by snake-biscuits** [comment]
Here's how to use the converter
```python
>>> from bsp_tool import ValveBsp
>>> from bsp_tool.branches import nexon
>>> bsp = ValveBsp(nexon.vindictus, "E:/Mod/Vindictus/xtruong712/w_boss04.bsp")
>>> from bsp_tool.extensions import convert
>>> convert.nexon.vindictus_steam_to_69(bsp)
```
converted maps will be saved to the folder this is run from
saving to another folder: `convert.nexon.vindictus_steam_to_69(bsp, outdir="E:/Mod/Vindictus/converted")`

don't forget to replace filenames with your own folders

**Comment by xtruong712** [comment]
> Here's how to use the converter
> 
> ```python
> >>> from bsp_tool import ValveBsp
> >>> from bsp_tool.branches import nexon
> >>> bsp = ValveBsp(nexon.vindictus, "E:/Mod/Vindictus/xtruong712/w_boss04.bsp")
> >>> from bsp_tool.extensions import convert
> >>> convert.nexon.vindictus_steam_to_69(bsp)
> ```
> 
> converted maps will be saved to the folder this is run from saving to another folder: `convert.nexon.vindictus_steam_to_69(bsp, outdir="E:/Mod/Vindictus/converted")`
> 
> don't forget to replace filenames with your own folders

thanks you, i will try it

**Comment by xtruong712** [comment]
@snake-biscuits 
![image](https://github.com/snake-biscuits/bsp_tool/assets/72967317/db9691db-4281-43e0-85e5-c2bef30af62c)
i got this error

**Comment by snake-biscuits** [comment]
Are you using the latest version of `bsp_tool` from GitHub?
`nexon.vinidictus69` is for supporting v1.69 and was added in May of this year
But the updates to work with the converter are from yesterday

The only other reason I can think of is overlapping `bsp_tool.branches.nexon` & `bsp_tool.extensions.convert.nexon`
But going by the error message it looks like both exist properly

**Comment by xtruong712** [comment]
> Are you using the latest version of from GitHub? is for supporting v1.69 and was added in May of this year But the updates to work with the converter are from yesterday`bsp_tool``nexon.vinidictus69`
> 
> The only other reason I can think of is overlapping & But going by the error message it looks like both exist properly`bsp_tool.branches.nexon``bsp_tool.extensions.convert.nexon`

i use version update yesterday

**Comment by xtruong712** [comment]
![image](https://github.com/snake-biscuits/bsp_tool/assets/72967317/2d26e123-6d42-4717-a4b6-836852ed5ece)
new error @@

**Comment by snake-biscuits** [comment]
That error means the Static Prop version isn't supported
```python
>>> from bsp_tool import ValveBsp
>>> from bsp_tool.branches import nexon
>>> bsp = ValveBsp(nexon.vindictus, "E:/Mod/Vindictus/hfs/2022/maps/4_01.bsp")
>>> bsp.loading_errors
{}
>>> bsp.GAME_LUMP.loading_errors
{}
>>> bsp.GAME_LUMP.headers["sprp"].version
7
```
^ this should give you something like:
```python
>>> bsp.GAME_LUMP.loading_errors
{"sprp": AttributeError(...)}
```
If `bsp.GAME_LUMP.headers["sprp"].version` is 7, you don't have the latest version of `bsp_tool/branches/nexon/vindictus.py`
if it's not 7, I need to extract the latest map from the `.hfs` and see if I can port that

**Comment by xtruong712** [comment]
![image](https://github.com/snake-biscuits/bsp_tool/assets/72967317/f475a4ea-156d-4bac-9025-b498363e87c7)
this is my error

**Comment by snake-biscuits** [comment]
3.11 should convert maps just fine.

**Comment by snake-biscuits** [comment]
Installing with pip would be very out of date
Unless you use:
`py -3.11 -m pip install git+https://github.com/snake-biscuits/bsp_tool.git`
This should install the latest version
The new version for pip won't be ready for a while yet

**Comment by xtruong712** [comment]
@snake-biscuits  hi, Nexon was update game lumps to version 8. now cant convert again. can you help me update that..

this is file new https://mega.nz/file/j95VSKjC#uvyQ0GBwQ-VeYRa-PeD05kpty21wd74Mi9y6A-0GKUU

**Comment by xtruong712** [comment]
@snake-biscuits  hope you can help ^^, sorry because borthing you

**Comment by xtruong712** [comment]
@snake-biscuits if have time, please help me 🥹

**Comment by snake-biscuits** [comment]
> hi, Nexon was update game lumps to version 8.

was able to throw together a pretty quick fix
looks like the `StaticProp` struct just got a little longer

the latest version of `bsp_tool` should be able to load the new format
it worked for the map you sent

let me know if you encounter any issues trying the latest version

**Comment by xtruong712** [comment]
extensiton/convert dont need update? 
@snake-biscuits

**Comment by snake-biscuits** [comment]
> extensiton/convert dont need update?

I had forgotten about that
Updated it now, and it seemed to work with the map you sent

Be sure to test it on your end and make sure it all looks good in-game

**Comment by MissWangyong** [comment]
old way cannot use
>>> from bsp_tool import ValveBsp
>>> from bsp_tool.branches import nexon
>>> bsp = ValveBsp(nexon.vindictus, "E:/Mod/Vindictus/xtruong712/w_boss04.bsp")
>>> from bsp_tool.extensions import convert
>>> convert.nexon.vindictus_steam_to_69(bsp)

**Comment by snake-biscuits** [comment]
did you get an error?
how did it fail to work?
which maps did you try it with?
do you have the latest version of `bsp_tool`?

**Comment by MissWangyong** [comment]
Traceback (most recent call last):
  File "E:\______Project_Hudaba\_python\python bsp ver678to5\est.py", line 8, in <module>
    convert.nexon.vindictus_steam_to_69(bsp)
  File "E:\______Project_Hudaba\_python\python bsp ver678to5\bsp_tool\extensions\convert\nexon.py", line 11, in vindictus_steam_to_69
    assert bsp.GAME_LUMP.headers["sprp"].version in (6, 7, 8)
AttributeError: 'ValveBsp' object has no attribute 'GAME_LUMP'
0

进程已结束,退出代码1

**Comment by MissWangyong** [comment]
from bsp_tool import ValveBsp
from bsp_tool.branches import nexon
from bsp_tool.extensions import convert
bsp = ValveBsp(nexon.vindictus, "./bsp/4_17.bsp")
print(bsp.version)
convert.nexon.vindictus_steam_to_69(bsp)

**Comment by snake-biscuits** [comment]
> AttributeError: 'ValveBsp' object has no attribute 'GAME_LUMP'

`convert.vindictus_steam_to_69` only converts the `GAME_LUMP`
If the map doesn't have a `GAME_LUMP` you shouldn't need to convert that map

**Comment by MissWangyong** [comment]
https://mega.nz/file/j95VSKjC#uvyQ0GBwQ-VeYRa-PeD05kpty21wd74Mi9y6A-0GKUU
Sorry I still can't get bsp_tool to work
Can you try to convert this file?

**Comment by xtruong712** [comment]
![Image](https://github.com/user-attachments/assets/a8c77faf-57dd-4632-9a7d-d3a5ea019c29)

this is problem now

**Comment by xtruong712** [comment]
This problem when i convert file w03, file i send you

**Comment by xtruong712** [comment]
> this is file new https://mega.nz/file/j95VSKjC#uvyQ0GBwQ-VeYRa-PeD05kpty21wd74Mi9y6A-0GKUU
> 

I use this file i send you before. And get some error like picture now. If have time please try it.
